### PR TITLE
chore(ClipboardCopy): updated tests

### LINKED ISSUE #9532: Clipboard copy - update unit tests

### PR DESCRIPTION
**Pull request by thatblindgeye**
**What**: Closes #9532 

**Additional issues**:

### CODE REVIEW

**Review by tlabaj — APPROVED** [review]

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyAction.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  );
+
+  expect(screen.getByTestId('container').firstChild).toBeVisible();
+});
+
+test(`Renders with class ${styles.clipboardCopyActionsItem} by default`, () => {
+  render(<ClipboardCopyAction>Action text</ClipboardCopyAction>);
+
+  expect(screen.getByText('Action text')).toHaveClass(styles.clipboardCopyActionsItem);
+});
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyAction.test.tsx:20]:
```suggestion

  expect(screen.getByText('Action text')).toHaveClass(styles.clipboardCopyActionsItem, { exact: true });
});
```

**Comment by thatblindgeye** [comment]
I also updated how the `aria-controls` within ClipboardCopyToggle was rendered, as it should only be the single `contentId` instead of the concatenated `id` + `contentId` (more of a bug fix than anything but can revert if necessary)

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  render(
+    <ClipboardCopy variant="expansion" toggleAriaLabel="toggle label">
+      {children}
+    </ClipboardCopy>
+  );
+
+  expect(screen.getByText('toggle-ariaLabel: toggle label')).toBeVisible();
+});
+
+test('Passes isReadOnly to TextInput', () => {
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx:253]: Can you add an inverse test for this?

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  expect(screen.getByRole('textbox')).toHaveAttribute('readonly');
+});
+
+test('Passes textAriaLabel to TextInput', () => {
+  render(<ClipboardCopy textAriaLabel="text label">{children}</ClipboardCopy>);
+
+  expect(screen.getByRole('textbox')).toHaveAccessibleName('text label');
+});
+
+test('Calls onChange', async () => {
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx:265]: Inverse test for this as well.

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+  render(<ClipboardCopy onChange={onChangeMock}>{children}</ClipboardCopy>);
+
+  await user.type(screen.getByRole('textbox'), typedText);
+
+  expect(onChangeMock).toHaveBeenCalledTimes(typedText.length);
+});
+
+test('Matches snapshot', () => {
+  const { asFragment } = render(<ClipboardCopy>{children}</ClipboardCopy>);
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx:278]: We probably want to add a fixed ouiaId here.

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  expect(asFragment()).toMatchSnapshot();
+});
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopy.test.tsx:280]: Can you add tests for the `onCopy` and `clickTip` props?

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx [hunk 6]
```diff
@@ -2,36 +2,126 @@ import React from 'react';
...
-  const user = userEvent.setup();
+test('Renders with aria-labelledby with passed id and textId prop values', () => {
+  render(
+    <>
+      <div id="text-id">Copyable text</div>
+      <ClipboardCopyButton {...requiredProps} />
+    </>
+  );
+
+  expect(screen.getByRole('button')).toHaveAttribute('aria-labelledby', 'button-id text-id');
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx:42]: WDYT about just checking the actual a11y behaviours for stuff like this? I'm not sure if doing the explicit attribute tests is actually adding value or if it's just tying us down more to our current implementation.

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx [hunk 7]
```diff
@@ -2,36 +2,126 @@ import React from 'react';
...
+  expect(screen.getByRole('button')).toHaveClass(buttonStyles.modifiers.control);
+});
+
+test('Passes variant to Button when variant is passed', () => {
+  render(<ClipboardCopyButton variant="plain" {...requiredProps} />);
+
+  expect(screen.getByRole('button')).toHaveClass(buttonStyles.modifiers.plain);
+});
+
+test('Calls onClick when passed', async () => {
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx:85]: Inverse test for this too please.

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx [hunk 8]
```diff
@@ -2,36 +2,126 @@ import React from 'react';
...
+test('Passes position to Tooltip', () => {
+  render(<ClipboardCopyButton position="bottom" {...requiredProps} />);
+
+  expect(screen.getByText('position: bottom')).toBeVisible();
+});
+
+test('Matches snapshot', () => {
+  const { asFragment } = render(<ClipboardCopyButton {...requiredProps} />);
+  expect(asFragment()).toMatchSnapshot();
 });
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyButton.test.tsx:127]: Can you add a test for `onTooltipHidden`?

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx [hunk 9]
```diff
@@ -1,48 +1,92 @@
...
-  test('has aria-expanded set to false when isExpanded is false', () => {
-    render(<ClipboardCopyToggle {...props} />);
+test('Spreads additional props to container', () => {
+  render(<ClipboardCopyToggle data-prop="test" {...requiredProps} />);
 
-    const toggleButton = screen.getByRole('button');
-    expect(toggleButton).toHaveAttribute('aria-expanded', 'false');
-  });
+  expect(screen.getByRole('button')).toHaveAttribute('data-prop', 'test');
 });
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx:92]: Can you add tests for the default class and the `className` prop?

**Review by wise-king-sullyman — COMMENTED** [review]
Awesome work on this so far, small handful of requsts/questions:

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx (hunk 9 above)
Comment (thatblindgeye) [reply packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx:92]: @tlabaj we had a question when discussing this. Currently the ClipboardCopyToggle component has `className` as part of the interface, but it's spread in the props object which is placed on our Button component internally. Do you happen to know if that was the intent (passing a className to ClipboardCopyToggle renders the class on `pf-v5-c-button`), or was the className meant to be passed on the `div.pf-v5-c-clipboard-copy__toggle-icon"`?

**Review by wise-king-sullyman — APPROVED** [review]
🥳 looks great other than that last question about className

File: packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx (hunk 9 above)
Comment (tlabaj) [reply packages/react-core/src/components/ClipboardCopy/__tests__/ClipboardCopyToggle.test.tsx:92]: Can open up a new breaking change issue to spread the className `div.pf-v5-c-clipboard-copy__toggle-icon`.  We may also consider adding a new prop to add a classname to the button.

**Pull request merged**